vendor/github.com/docker/distribution/registry/storage/linkedblobstore.go

```diff
@@ -289,11 +281,19 @@ func (lbs *linkedBlobStore) Enumerate(ctx context.Context, ingestor func(digest.
 	return nil
 }
 
-func (lbs *linkedBlobStore) mount(ctx context.Context, sourceRepo reference.Named, dgst digest.Digest) (distribution.Descriptor, error) {
+func (lbs *linkedBlobStore) mount(ctx context.Context, repoOverride distribution.RepositoryOverride, sourceRepo reference.Named, dgst digest.Digest) (distribution.Descriptor, error) {
 	repo, err := lbs.registry.Repository(ctx, sourceRepo)
 	if err != nil {
 		return distribution.Descriptor{}, err
 	}
+
+	if repoOverride != nil {
+		repo, err = repoOverride(ctx, repo)
```

Comment:
see the last couple commits in https://github.com/liggitt/origin/commits/miminar-remote-layer-federation as an example. That keeps the flow of information going in one direction, and I'd prefer a smaller upstream change like that